api/v1alpha1/well_known_annotations.go

```diff
@@ -3,4 +3,5 @@ package v1alpha1
 const (
 	AnnotationAutoUpdate     = "packages.glasskube.dev/auto-update"
 	AnnotationInstalledAsDep = "packages.glasskube.dev/installed-as-dependency"
+	AnnotationSpecHash       = "packages.glasskube.dev/spec-hash"
```

Comment:
I think something like `package-spec-hashed` or `package-spec-sha256` would be a little more meaningful to someone unfamiliar with the internal mechanics